python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -109,6 +109,11 @@ def test_all_none_category(self):
         df['a'] = df['a'].astype('category')
         self._check_pandas_roundtrip(df)
 
+    def test_non_string_columns(self):
```

Comment:
There should be a test for additional column types that either fails or explicitly succeeds based on what we decide about allowing other types in.